### PR TITLE
Functional Component Refactor: LinkToolbar

### PR DESCRIPTION
**Pull request by ItaloSa**
Issue: #4204.
Refactoring component to functional.

### CODE REVIEW

File: app/editor/components/LinkToolbar.tsx [hunk 1]
```diff
@@ -31,39 +31,34 @@ function isActive(props: Props) {
...
+    },
+    [menuRef, props]
+  );
+
+  React.useEffect(() => {
+    window.addEventListener("mousedown", handleClickOutside);
+    return () => window.removeEventListener("mousedown", handleClickOutside);
+  }, [handleClickOutside]);
+
+  const handleOnCreateLink = async (title: string) => {
```
Comment (tommoor) [inline app/editor/components/LinkToolbar.tsx:57]: This seems to have broken the functionality `cmd`+`k` on an empty line no longer brings up the link insert toolbar. I suspect that these two callbacks need to be memoized with `useCallback`

**Review by tommoor — CHANGES_REQUESTED** [review]

File: app/editor/components/LinkToolbar.tsx (hunk 1 above)
Comment (ItaloSa) [reply app/editor/components/LinkToolbar.tsx:57]: Just to be clear, are the two callbacks you've said `handleOnCreateLink` and `handleOnSelectLink`?

Comment (tommoor) [reply app/editor/components/LinkToolbar.tsx:57]: Yes, anything that isn't memo'd

**Comment by tommoor** [comment]
Merged in ec2da746d – needed some work still, but gave co-authorship credit

**Pull request closed without merging**